zha/application/platforms/light/__init__.py

```diff
@@ -1064,6 +1093,18 @@ def _assume_group_state(self, update_params) -> None:
 
             self.maybe_emit_state_changed_event()
 
+    def restore_external_state_attributes(
+        self,
+        *,
+        state: bool | None,
+        off_with_transition: bool | None,
+        off_brightness: int | None,
+    ) -> None:
+        """Restore extra state attributes that are stored outside of the ZCL cache."""
+        self._state = state
```

Comment:
When `state` is passed from HA to this, we'd need to make sure it's actually a `bool | None`, not a `str`. 
HA uses `"off"`, `"on"`, `"unknown"`, `"unavailable"`. So, [`STATE_ON`, ...](https://github.com/home-assistant/core/blob/61f1c8d9631bab1a4236eee405c5b0b03e8be22e/homeassistant/const.py#L463-L492)

For reference, current code for restoring on/off state in HA Core ZHA [here.](https://github.com/home-assistant/core/blob/61f1c8d9631bab1a4236eee405c5b0b03e8be22e/homeassistant/components/zha/light.py#L849).

Reply:
It's converted into a `bool` within the Core code: <https://github.com/home-assistant/core/blob/27bd52d26572b4a7d329c44a1b40a6271c5af6d9/homeassistant/components/zha/light.py#L166-L168>. I'm hoping static type checking will pick up invalid invocations.